pkg/agent/datamodel/helper.go

```diff
@@ -93,3 +94,25 @@ func IndentString(original string, spaces int) string {
 	}
 	return out.String()
 }
+
+func processContainerImage(downloadURL string) (string, error) {
+	// example URL "downloadURL": "mcr.microsoft.com/oss/kubernetes/autoscaler/addon-resizer:*",
+	// getting the data between the last / and the last :
+	parts := strings.Split(downloadURL, "/")
+	if len(parts) == 0 || len(parts[len(parts)-1]) == 0 {
+		return "", fmt.Errorf("container image component URL is not in the expected format: %s", downloadURL)
+	}
+	lastPart := parts[len(parts)-1]
+	component := strings.TrimSuffix(lastPart, ":*")
+	return component, nil
+}
+
+func processDownloadFile(downloadURL string) (string, error) {
+	// example URL "downloadURL": "https://acs-mirror.azureedge.net/cni-plugins/v*/binaries",
+	url, err := url.Parse(downloadURL) // /cni-plugins/v*/binaries
+	if err != nil {
+		return "", fmt.Errorf("download file image URL is not in the expected format: %s", downloadURL)
+	}
+	urlSplit := strings.Split(url.Path, "/") // ["", cni-plugins, v*, binaries]
+	return urlSplit[1], nil
```

Comment:
lets return an error here if there aren't at least 2 elements in the URL split, or if the segment we want is empty